### DISCUSSION TITLE
How can I generate a sidebar for an API that doesn't have tags?

### DISCUSSION THREAD

**Discussion by gsejourne** [PaloAltoNetworks/docusaurus-openapi-docs#834] [Q&A]
First, I've been using the plugin for some time, and it works great.
However, some of my API definitions don't have tags on the endpoints.
I tried multiple things to generate the sidebar anyway, WITHOUT grouping the endpoints by tags, but it doesn't work:
- If I don't add the `sidebarOptions` property at all, it does **NOT** generate the `sidebar.js` file altogether
- if I add a `sidebarOptions.groupPathsBy = ''` it fails with `[Error [ValidationError]: "config.bills.sidebarOptions.groupPathsBy" must be one of [tag, tagGroup]]`
- if I add an empty `sidebarOptions` property to the API config, it builds fine, but again, it does **NOT** generate the `sidebar.js` file.

Any help appreciated! 
Thanks.

**Comment by sserrata** [comment]
Hi @gsejourne, curious if you've read the docs? If you have, is there anything missing that would otherwise have helped? Thanks!

https://docusaurus-openapi.tryingpan.dev/sidebars

**Reply by gsejourne** [reply]
Hi @sserrata 
Yes, I have gone through the docs, which details how to use tags, how you should name them, etc. Unfortunately, it doesn't describe what to do in case you **don't** want to use tags.
Typically, the doc [here](https://docusaurus-openapi.tryingpan.dev/#sidebaroptions) states:

> Currently, groupPathsBy only contains support for grouping by tag.

The question is, does it support NOT grouping at all ;)
Thanks

**Reply by sserrata** [reply]
> if I add an empty sidebarOptions property to the API config, it builds fine, but again, it does NOT generate the sidebar.js file.

Hi @gsejourne, perhaps you overlooked this section? https://docusaurus-openapi.tryingpan.dev/sidebars#built-in-docusaurus-sidebars

The docs generated by this plugin are compatible with Docusaurus sidebars so, if you're not grouping by tag, the alternative is to use `autogenerated` or create the sidebar manually. You can even define your own sidebars generator function if needed: https://docusaurus.io/docs/sidebar/autogenerated#customize-the-sidebar-items-generator

**Reply by gsejourne** [reply]
Thank you @sserrata 
Indeed, I initially tried reverting to docusaurus `autogenerated` sidebars, but for some reason, the endpoints are not in order (most likely because it's taken from the object list of endpoints in the YAML... don't know.
In any case, I'll try my own sidebar generator function and find something that works.
Thanks for your help!
